universal-login-relayer/lib/services/messages/PendingMessages.ts

```diff
@@ -1,62 +1,68 @@
-import {Wallet} from 'ethers';
-import {calculateMessageHash, SignedMessage, INVALID_KEY} from '@universal-login/commons';
+import {Wallet, Contract} from 'ethers';
+import {calculateMessageHash, concatenateSignatures, SignedMessage, INVALID_KEY} from '@universal-login/commons';
+import WalletContract from '@universal-login/contracts/build/WalletMaster.json';
 import {DuplicatedSignature, InvalidSignature, DuplicatedExecution, InvalidTransaction, NotEnoughSignatures} from '../../utils/errors';
-import {getKeyFromHashAndSignature} from '../../utils/utils';
-import PendingMessage from './PendingMessage';
-import IPendingMessagesStore from './IPendingMessagesStore';
+import IPendingMessagesStore, {CollectedSignatureKeyPair} from './IPendingMessagesStore';
+import {getKeyFromHashAndSignature, sortSignatureKeyPairsByKey, newPendingMessage} from '../../utils/utils';
 
 export default class PendingMessages {
 
   constructor(private wallet : Wallet, private messagesStore: IPendingMessagesStore) {
   }
 
-  isPresent(messageHash : string) {
+  async isPresent(messageHash : string) {
```

Comment:
No need this function. You can inline it, that is simply use:
`this.messagesStore.isPresent(hash)` everywhere instead of: `this.isPresent(hash) `

Reply:
MessageHandler use this function in ```ts async getStatus(hash: string) {
    if (await this.pendingMessages.isPresent(hash)){ ```